packages/aws-cdk-lib/aws-kms/lib/key.ts

```diff
@@ -434,6 +434,13 @@ export interface KeyProps {
    */
   readonly enableKeyRotation?: boolean;
 
+  /**
+   * The period between each automatic rotation.
+   *
+   * @default - 365 days.
+   */
+  readonly rotationPeriod?: Duration;
```

Comment:
I've simply added `rotationPeriod` properties to `KeyProps`.

However, there are other possible implementation approaches, and I'm struggling to decide which one is best. Could I please get your opinions, reviewers?


1. add rotationPeriod property (current implementation)

```ts
new kms.Key(this, 'MyKey', {
  enableKeyRotation: true,
  rotationPeriod: Duration.days(180), // Add
});
```
3. deprecate `enableKeyRotation` and define `KeyRotation` property
```ts
export interface KeyRotation {
  enableKeyRotation: boolean,
  rotationPeriod?: Duration,
}

new kms.Key(this, 'MyKey', {
  keyRotation: {
    enableKeyRotation: true,
    rotationPeriod: Duration.days(180), // optional
  },
});
```

5. deprecate `enableKeyRotation` and add only `rotationPeriod`

```ts
new kms.Key(this, 'MyKey', {
  rotationPeriod: Duration.days(180), // Implicitly enable key rotation by defining this property
});
```

Reply:
I don't think the current implementation is that terrible. It probably wouldn't have been my first choice if `enableKeyRotation` wasn't already there, but given the circumstances, you're not introducing a deprecation or breaking change, you're allowing the existing construct to customize its default value.

Reply:
@nmussy Thank you for your opinion. I'm relieved to hear that.

Reply:
I like the implementation you chose @badmintoncryer, just with a small modification; if `rotationPeriod` is defined and `enableKeyRotation` is `undefined`, let's set `enableKeyRotation` to `true`.